snapshot-labs/sx-monorepo
# fix: disable sidebar swipping on large screen

### Summary



Closes: #741 

This PR disable the right swipping to open the sidebar on large screen, where the sidebar is already visible.

Swipping right is:

- showing the sidebar
- disable the page scroll
- show the backdrop

Currently, on large screen, since the sidebar is already visible, it just disable the page scroll, with the backdrop also hidden on large screen, giving a non-scrollable page without any indication.

This PR:

- disable the sidebar swipping on large screen
- keep the backdrop on large screen, so that when resizing from a small screen with a `sidebarOpen` state to a large screen, we have a visual feedback about the `sidebarOpen` state, and have a way to exit it
- 

### How to test

1. On a large screen, try swipping right
2. It should do nothing (no backdrop, page still scrollable)
3. On a small screen, try swipping right
4. It should open the sidebar, and show the backdrop
5. Clicking on the backdrop or swipping left should close the sidebar
6. On a small screen with the sidebar open, resize the screen to a large screen
7. The sidebar should still be open, and backdrop still here
8. Clicking on the backdrop should exit the `sidebarOpen` state, and make the page scrollable again
